Sources/MongoSwift/BSON/DocumentIterator.swift

```diff
@@ -190,6 +212,31 @@ public class DocumentIterator: IteratorProtocol {
 #endif
     }
 
+    /// Internal helper function for passing an array of strings from Swift to C (i.e. char **)
+    internal static func withVaListOfCStrings(
```

Comment:
some API design related nits:

1) when I suggested the name, I hadn't considered that we could move the call to copy in here too, but it definitely makes sense to do so. given that this does more than just create a va list of c strings now, I think we could name this better to reflect what is happening -- maybe even just `copy` or `copyElements` or something like that.

2) in methods where what the parameters actually mean are ambiguous, parameter labels are very useful for someone reading usages of the method later on. here we are currently not requiring them by putting a `_` before each one. if you haven't yet, I would recommend reading over the [Swift API Design Guidelines](https://swift.org/documentation/api-design-guidelines/) for some guidance on this. it's a Swift-y API pattern to make function calls read like phrases/sentences, so we should strive for something that reads along the lines of "copy from x to y excluding z".

3) Do we need to be able to specify `firstExclude` separately, or could we just take in all the ones to exclude in `args` and automatically use the first one as `firstExclude`?

Reply:
1. I changed the name to `copyElements` as requested.
2. Thanks for this - I wasn't completely sure what the parameter labels were doing in other functions so this is very helpful!
3. I've removed `firstExclude` and the helper function now throws if no keys are provided to exclude. I chose this as one should be using the C function `bson_copy` if this were the case.